.github/workflows/shell.yml

```
@@ -48,10 +48,13 @@ jobs:
   lint_shfmt:
     name: Shell - lint_shfmt
     runs-on: ubuntu-20.04
-    container:
-      image: ghcr.io/build-trust/ockam-builder@sha256:5ab42598e35509cad3ea9c1e1bd0ed135ed1340c6ae44b762b1c8bbec31d5c68
     steps:
       - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab
         with:
           ref: ${{ github.event.inputs.commit_sha }}
-      - run: shfmt -d $(find . -not -path "./demos/*" -iname "*.sh" -o -name "*.bash")
+
+      - name: Install Nix
+        uses: cachix/install-nix-action@5cfd5166ea7680b784fd8cf556c6d07a9ecce260
+
+      - name: Setup Ockam Nix Environment
+        run:  nix develop ./tools/nix#tooling --ignore-environment --command shfmt -d $(find . -not -path "./demos/*" -iname "*.sh" -o -name "*.bash")
```

In some cases like this one, I would consider abstracting this further:

In `parts/tooling.nix`:
```nix
    packages.shfmt-all = pkgs.writeShellApplication {
      name = "shfmt-all";
      runtimeInputs = with pkgs; [findutils shfmt];
      text = ''
        find . -not -path "./demos/*" \
          -a \( -iname "*.sh" -o -name "*.bash" \) \
          -exec shfmt --diff {} +
      '';
    };
```

```suggestion
      - name: Setup Ockam Nix Environment
        run:  nix run ./tools/nix#shfmt-all
```

```

https://nixos.org/manual/nixpkgs/stable/#trivial-builder-writeShellApplication

> Unlike with normal writeShellScriptBin, there is no need to manually write out ${curl}/bin/curl, setting the PATH was handled by writeShellApplication. Moreover, the script is being checked with shellcheck for more strict validation. 

The included shellcheck pass actually helps me catch minor stuff pretty often.